hack/bootstrap-kubestellar.sh

```diff
@@ -0,0 +1,262 @@
+#
+# Arguments:
+# [--kcp-version release] set a specific kcp release version, default: latest
+# [--kubestellar-version release] set a specific KubeStellar release version, default: latest
+# [--os linux|darwin] set a specific OS type, default: autodetect
+# [--arch amd64|arm64] set a specific architecture type, default: autodetect
+# [--folder name] sets the installation folder, default: $PWD/kcp-edge
+# [--bind address] bind kcp to a specific ip address
+# [--ensure-imw name] create a Inventory Management Workspace (IMW)
+# [--ensure-wmw name] create a Workload Management Workspace (WMW)
+# [-V|--verbose] verbose output
+
+set -e
+
+kcp_installed() {
+    if [ "$(which kcp)" == "" ]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kcp_running() {
+    if [ "$(pgrep -f 'kcp start')" == "" ]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kcp_get_latest_version() {
+    curl -sL https://github.com/kcp-dev/kcp/releases/latest | grep "</h1>" | head -n 1 | sed -e 's/<[^>]*>//g' | xargs
+}
+
+kubestellar_installed() {
+    if [[ "$(which mailbox-controller)" == "" || "$(which placement-translator)" == "" || "$(which scheduler)" == "" ]]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kubestellar_running() {
+    if [[ "$(pgrep -f mailbox-controller)" == "" ||  "$(pgrep -f placement-translator)" == "" || "$(pgrep -f 'scheduler -v')" == "" ]]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kubestellar_get_latest_version() {
+    curl -sL https://github.com/kcp-dev/edge-mc/releases/latest | grep "</h1>" | head -n 1 | sed -e 's/<[^>]*>//g' | xargs
+}
+
+get_os_type() {
+  case "$OSTYPE" in
+      linux*)   echo "linux" ;;
+      darwin*)  echo "darwin" ;;
+      *)        echo "Unsupported operating system type: $OSTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_arch_type() {
+  case "$HOSTTYPE" in
+      x86_64*)  echo "amd64" ;;
+      aarch64*) echo "arm64" ;;
+      *)        echo "Unsupported architecture type: $HOSTTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_full_path() {
+    echo "$(cd "$1"; pwd)"
+}
+
+kcp_version=""
+kubestellar_version=""
+os_type=""
+arch_type=""
+folder=""
+verbose=""
+kcp_address=""
+kubestellar_imw=""
+kubestellar_wmw=""
+
+while (( $# > 0 )); do
+    case "$1" in
+    (--kcp-version)
+        if (( $# > 1 ));
+        then { kcp_version="$2"; shift; }
+        else { echo "$0: missing release version" >&2; exit 1; }
+        fi;;
+    (--kubestellar-version)
+        if (( $# > 1 ));
+        then { kubestellar_version="$2"; shift; }
+        else { echo "$0: missing release version" >&2; exit 1; }
+        fi;;
+    (--os)
+        if (( $# > 1 ));
+        then { os_type="$2"; shift; }
+        else { echo "$0: missing OS type" >&2; exit 1; }
+        fi;;
+    (--arch)
+        if (( $# > 1 ));
+        then { arch_type="$2"; shift; }
+        else { echo "$0: missing architecture type" >&2; exit 1; }
+        fi;;
+    (--bind)
```

Comment:
Maybe the name should be a little more specific, so a reader of the command line gets a bit of a clue what it is for?